api/desecapi/tests/testrrsets.py

```diff
@@ -189,6 +190,107 @@ def test_create_my_rr_sets(self):
                 self.assertStatus(response, status.HTTP_200_OK)
                 self.assertRRSet(response.data, **data)
 
+    def test_bulk_post_my_rr_sets(self):
+        data = []
+        for subname in ['', 'create-my-rr-sets', 'foo.create-my-rr-sets', 'bar.baz.foo.create-my-rr-sets']:
+            data.extend([
+                {'subname': subname, 'records': ['1.2.3.4'], 'ttl': 60, 'type': 'A'},
+                {'subname': subname, 'records': ['desec.io.'], 'ttl': 900, 'type': 'PTR'},
+            ])
+
+        with self.assertPdnsRequests(self.requests_desec_rr_sets_update(name=self.my_empty_domain.name)):
+            response = self.client.bulk_post_rr_sets(domain_name=self.my_empty_domain.name, payload=data)
+            self.assertStatus(response, status.HTTP_201_CREATED)
+
+        response = self.client.get_rr_sets(self.my_empty_domain.name)
+        self.assertStatus(response, status.HTTP_200_OK)
+        self.assertRRSetsCount(response.data, data)
+
+        for value in data:
+            response = self.client.get_rr_set(self.my_empty_domain.name, value['subname'], value['type'])
+            self.assertStatus(response, status.HTTP_200_OK)
+            self.assertRRSet(response.data, **value)
+
+    def test_bulk_patch_my_rr_sets(self):
+        data_full = [
+            {'subname': 'my-patch', 'records': ['1.2.3.4'], 'ttl': 60, 'type': 'A'},
+            {'subname': 'my-patch', 'records': ['desec.io.'], 'ttl': 60, 'type': 'PTR'},
+        ]
+
+        data_no_records = copy.deepcopy(data_full)
+        data_no_records[1].pop('records')
+
+        data_no_ttl = copy.deepcopy(data_full)
+        data_no_ttl[0].pop('ttl')
+
+        data_no_type = copy.deepcopy(data_full)
+        data_no_type[1].pop('type')
+
+        # Fresh records need TTL and type
+#        with self.assertPdnsNoRequestsBut():
+#            response = self.client.bulk_patch_rr_sets(domain_name=self.my_empty_domain.name, payload=data_no_records)
+#            self.assertStatus(response, status.HTTP_400_BAD_REQUEST)
+#            self.assertEqual(response.data, [{'ttl': ['This field is required for new RRsets.']}, {}])
+        with self.assertPdnsNoRequestsBut():
+            response = self.client.bulk_patch_rr_sets(domain_name=self.my_empty_domain.name, payload=data_no_ttl)
+            self.assertStatus(response, status.HTTP_400_BAD_REQUEST)
+            self.assertEqual(response.data, [{'ttl': ['This field is required for new RRsets.']}, {}])
```

Comment:
maybe we should transform these kind of statements into `assertData` or `assertBody` similar to what we did with `assertStatus`

Reply:
sure, but I think this is independent of adding bulk request tests